[J4] Deleting jumps in SystemPanel

korenevskiy opened the pull request:
Colleagues, I did not delete Masonry. I added a few properties to test the absence of page jumps.

I ask you to test these changes and compare them with RC. You will see the superiority.

The Masonry layout implies automatic placement depending on the height of the block. So that the lower border of the parent's space is aligned with the content. Which leads to jumping blocks.
Thus, the order of the blocks was imposed on us by the JS Masonry script. At your discretion. But in the cards there were blocks in which the number of elements is the same. Even they can't be swapped at our discretion. The JS script has its own logic.
I turned off Masonry. Sorted the blocks as the creator/maintainer of the sites.
As a result, it was possible to remove the block jumps when the page was loading.
Please test the PR

Please test it, and I need advice on what to remove from the unnecessary properties of the previous styles.

korenevskiy committed:
Update system.xml

korenevskiy committed:
Update _global.scss

korenevskiy committed:
Update _variables.scss

korenevskiy committed:
Update _global.scss

korenevskiy committed:
Update _global.scss

brianteeman commented:
I have tested this item :white_check_mark: successfully on 05a2e739d5c64d905109632e878d42d30347aff7<br /><br />I have no opinion about the change in order but the "jump" has indeed stopped<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35040">issues.joomla.org/tracker/joomla-cms/35040</a>.</sub>

korenevskiy commented:
> I have tested this item ✅ successfully on [05a2e73](https://github.com/joomla/joomla-cms/commit/05a2e739d5c64d905109632e878d42d30347aff7)
> 
> I have no opinion about the change in order but the "jump" has indeed stopped
> 
> This comment was created with the [J!Tracker Application](https://github.com/joomla/jissues) at [issues.joomla.org/tracker/joomla-cms/35040](https://issues.joomla.org/tracker/joomla-cms/35040).

There are properties in the card styles
```
column-count: 3;
column-gap: 1.25rem;
orphans: 1;
```
and
```
display: grid;
grid-gap: 0 15px;
grid-template-columns: 1fr;
```
these styles are applied to the same parent, but they override each other.
I remember last year the first type of properties was used, but now the second properties are active.

Both properties are available in styles. Let's delete the second one, and we will return to the previous layout of the card placements.

richard67 commented:
@korenevskiy Please fix the code style errors reported by drone. You can find them by following the "Details" link right beside the failed build at the bottom of the PR and then in the tree at the left hand side check for failed test in red colour. Or follow this link here: https://ci.joomla.org/joomla/joomla-cms/46356/1/20 .

korenevskiy committed:
Update _global.scss

korenevskiy committed:
Update _global.scss

korenevskiy committed:
Update _global.scss

korenevskiy committed:
Update _global.scss

korenevskiy commented:
This fix causes an error on the add a new module page.

![image](https://user-images.githubusercontent.com/6898474/128181067-fd7b0ab3-2c2e-4704-934a-f4bfe0c83b58.png)
Styles need to be finalized.

korenevskiy committed:
Update _com_modules.scss

korenevskiy commented:
I fixed the error in the modules.
As I find such errors, I will correct them.

korenevskiy commented:
![image](https://user-images.githubusercontent.com/6898474/128182906-aeb35013-d856-42fd-a3cf-6fad54b2a1a5.png)

korenevskiy commented:
![image](https://user-images.githubusercontent.com/6898474/128183489-d4b1fc02-5484-421b-9545-81d954002e0f.png)
Please show a screenshot with the full list of models

korenevskiy committed:
Update _com_modules.scss

korenevskiy committed:
Update _com_modules.scss

korenevskiy commented:
There is a rule there.
For there to be an analog of masonry, the parent must have the DISPLAY: BLOCK property;
To make it all the same: DISPLAY: GRID;
it is necessary to alternate these 2 properties depending on what is desired.

korenevskiy commented:
![image](https://user-images.githubusercontent.com/6898474/128186597-cec55b8e-383e-4e17-827e-df1599b823bb.png)

korenevskiy committed:
Update _global.scss

korenevskiy committed:
Update _global.scss

korenevskiy committed:
Update _global.scss

korenevskiy commented:
The drone hovered.

Quy commented:
The Home Dashboard is affected.

![35040](https://user-images.githubusercontent.com/368084/128214104-2f1d84c0-22ad-4c2b-8a9a-62036f85b976.jpg)

korenevskiy commented:
> The Home Dashboard is affected.
> 

Not happening. I have a normal display.
I will not be able to fix it, since everything is fine with me.
Please look at the reason for this.

C-Lodder commented:
Are you sure the grid gap still works in Safari?

korenevskiy commented:
> Are you sure the grid gap still works in Safari?

I am not sure of anything, since I have no way to believe it, and I have never been able to check it before. 
But I didn't change anything in the code.
Except for DISPLAY: GRID; changed to DISPLAY:BLOCK;
In fact, if it worked before, it works now, and if it didn't work before in SAFARI, it doesn't work now. I didn't add any new technologies, I just fixed a bug.
If you help us check this gap, the society will be grateful.

C-Lodder commented:
> In fact, if it worked before, it works now

Not necessarily. If I rightly remember, `grid-gap` doesn't work with multi column layouts....which you've added (`column-count`).
I dont have Safari, but would suggest it as part of the test criteria.

Quy commented:
Before PR:
![35040-before](https://user-images.githubusercontent.com/368084/128379411-9b1b8fb1-2104-4155-a2b6-94ec475a408c.jpg)

After PR:
![35040-after](https://user-images.githubusercontent.com/368084/128379423-bb585ad1-f952-4c1a-abd3-180ee05282d9.jpg)

korenevskiy commented:
@Quy  
I absolutely agree with you. This ugliness of sorting needs to be corrected. I was going to make this sorting to another PR.
The fact is that this order is determined by the order values in the database for modules.
Even if we make changes to the SQL files in the distribution. Then these changes will appear only with a fresh Joomla installation.
.
In your case, this is configured in the module manager. You need to open the list of modules for the cPanel home page, and then configure the order with the mouse.

korenevskiy committed:
Update _global.scss

korenevskiy committed:
Update _global.scss

korenevskiy committed:
Update _global.scss

korenevskiy committed:
Update _global.scss

korenevskiy commented:
I think 3 columns is good.

I found a problem.
I wanted to make it possible to use 3 options:
1 column, 2 columns, 3 columns, depending on the width of the screen.
But when trying to make amendments to
[`atum/scss/blocks/_global.scss`](https://github.com/joomla/joomla-cms/blob/d2322b3221920b6d471f22ddfb3bd2a8fe0b04eb/administrator/templates/atum/scss/blocks/_global.scss)

I saw the style the style
[`atum/scss/vendor/bootstrap/_card.scss`](https://github.com/korenevskiy/joomla-cms/blob/Deleting-jumps-Joom4/administrator/templates/atum/scss/vendor/bootstrap/_card.scss)

It overrides the template styles with its properties.
In fact, probably, the order should be reversed, the template styles should override the Bootstrap style for their needs.

How to make a 2-column mode for medium screens?

korenevskiy commented:
It seems that it is necessary to redefine the columns specified by  [`atum/scss/blocks/_global.scss`](https://github.com/joomla/joomla-cms/blob/d2322b3221920b6d471f22ddfb3bd2a8fe0b04eb/administrator/templates/atum/scss/blocks/_global.scss)
here 
[`atum/scss/pages/_com_cpanel.scss`](https://github.com/korenevskiy/joomla-cms/blob/Deleting-jumps-Joom4/administrator/templates/atum/scss/pages/_com_cpanel.scss)

korenevskiy committed:
Update _com_cpanel.scss

korenevskiy committed:
Update _com_cpanel.scss

Fedik commented:
Sorry, this is "No go".

Basically what you did here:  changing the flow from `left => right` to `top => bottom`. 
That throw us back to state when it was implemented first, and makes an issue for future ordering by User.

To eliminate jumps, all AJAX call from quickicon plugins must be done after `load`, not while `DOMContentLoaded` (as it currently) that extends "page loading time" and cause visible jumps.

Do not lose your time here.

korenevskiy committed:
Update _com_cpanel.scss

richard67 commented:
@korenevskiy Please check the above comment https://github.com/joomla/joomla-cms/pull/35040#issuecomment-894101687 and consider closing this PR. I think too that it goes the wrong way.

korenevskiy committed:
Update _com_cpanel.scss

korenevskiy committed:
Update _global.scss

korenevskiy committed:
Update _com_cpanel.scss

korenevskiy committed:
Update _com_cpanel.scss

korenevskiy commented:
I'm a little confused with the styles. Since there is no experience in the SCSS preprocessor.
In the [administrator/templates/atomic/css/pages/_com_cpanel.css](https://github.com/joomla/joomla-cms/blob/63343b890d33c5386d0adb293fc26f7dc31c030a/administrator/templates/atum/scss/pages/_com_cpanel.scss) styles
in line 69, the styles are not compiled to CSS.
Can you tell me what's the matter?

I want to make the layout of the main page have not 2 column modes, but 3.
Previously, there were modes: 1 column and 3 columns.
I want to make it so that there is: 1 column, 2 columns, and 3 columns.

Late loading of AJAX modules can be done in another way. Load it inside the TEMPLATE tag, and then expand the TEMPLATE after loading the JS.

korenevskiy committed:
Update _com_cpanel.scss

korenevskiy commented:
Please take a look,
There is CSS Masonry here
[`https://codepen.io/collection/XPjvPP/`](https://codepen.io/collection/XPjvPP/)

korenevskiy committed:
Update _com_cpanel.scss

korenevskiy committed:
Update _com_cpanel.scss

chmst commented:
I close this now, thank you @korenevskiy for your work and time, but at the moment this is a waste of time for all.

richard67 commented:
See #35069 .
